src/main.js

```diff
@@ -36,24 +35,24 @@ Apify.main(async () => {
             const { url } = context.request;
             log.info('Page opened.', { url });
 
-            await scrapeGithubIssues(context, issuesState);
+            await scrapeGithubIssues(context, repositoriesState);
         },
     });
 
     log.info('Starting Github issues crawl.');
     await crawler.run();
     log.info('Crawl finished.');
 
-    await Apify.setValue(ISSUES_STATE, issuesState);
+    await Apify.setValue(REPOSITORIES_STATE, repositoriesState);
 
     const issuesStore = await Apify.openKeyValueStore(ISSUES_KEY_VALUE_STORE);
-    const previousState = await issuesStore.getValue(ISSUES_STATE);
+    const previousState = await issuesStore.getValue(REPOSITORIES_STATE);
 
     if (!previousState) {
         log.info('No previous state of GitHub issues found in global key value store github-issues.');
         log.info('Saving the current state without comparing to the previous state. No Slack notification will be send.');
     } else {
-        const modifiedIssues = getModifiedIssues(issuesState, previousState);
+        const modifiedIssues = getModifiedIssues(repositoriesState, previousState);
```

Comment:
Sorry if Im' missing something but I don't see the `repositoriesState` being populated with old issues so that in case of some missed, it won't just fire the notifications again. Am I missing it somewhere? From my view, it would be the simplest to add it here:

```
const repositoriesState = await Apify.getValue(REPOSITORIES_STATE) || {};
```
Instead of `{}` it would load the named KV general state. (The 2 state formats should be ideally the same)

Reply:
@metalwarrior665  I wanted to keep scraping issues phase apart from the issues comparison and notification phases, would it be possible using your approach? I thought that the two phases should be independent, we'd scrape all issues from GitHub during the scraping phase and once the crawler finishes, we'd compare them with the previous state. Using `const repositoriesState = await Apify.getValue(REPOSITORIES_STATE) || previousState`, wouldn't we have to send notification right after we scrape a new issue and recognize it as a modified one? It should work fine but notifications would be handled directly while scraping.

The main PR change is that named KVS doesn't get overwritten by local KVS at the end of the script. It is only updated by adding new issues (detected during the comparison of named KVS from the previous run and local KVS from the current run). That's what `saveRepositoryUpdates` function is meant for - it should save new issues into named KVS and update the old ones if their state changed. Once we have some issues stored in the named KVS, they stay there forever so even if they're not scraped in some run (because of the migration), they won't be erased and detected as new ones in the following run.

Reply:
Ok, that sounds good :) Maybe just add some comments to the code, it is fairly complicated and I didn't understand it right away :) And then we can merge :)